app/updatemanager.go

```diff
@@ -301,11 +319,66 @@ func (c *ControlMap) Set(cm *UpdateControlMap) {
 			return
 		}
 	}
+	// If the map exists in the expired pool, the map is removed from the
+	// expired pool
+	if UpdateManagerMapExpired.Contains(cm) {
+		UpdateManagerMapExpired = UpdateManagerMapExpired.Delete(cm)
+	}
+	// Make sure the Expired map has the same keys
+	UpdateManagerMapExpired.InsertKey(cm.ID)
 	c.controlMap[cm.ID] = append(c.controlMap[cm.ID], cm)
 }
 
 func (c *ControlMap) Get(ID string) []*UpdateControlMap {
 	c.mutex.Lock()
 	defer c.mutex.Unlock()
-	return c.controlMap[ID]
+	m := c.controlMap[ID]
+	// Prune the control map based on the UpdateCOntrolMapTimeout expiry
+	// FIXME - This should probably be handled some other way, like a
+	// separate goroutine (?)
+	for i, e := range m {
```

Comment:
This probably needs to be separated out, so that the expired map does not check it's own expiries